modules/agar/src/main/ts/ephox/agar/assertions/ApproxStructures.ts

```diff
@@ -234,6 +265,22 @@ const assertAttrs = (expectedAttrs: Record<string, StringAssert>, actual: SugarE
   });
 };
 
+const assertAttrsWithSpecifics = (expectedAttrs: Record<string, StringAssert>, actual: SugarElement<Element>) => {
+  const allDefinedAttrs = Obj.keys(expectedAttrs);
+  const actualDefinedAttrs = Obj.keys(Obj.bifilter(Attribute.clone(actual), (_value, key) => !Arr.contains([ 'style', 'class' ], key)).t);
+
+  const isEqual = Arr.equal(
+    Arr.sort(allDefinedAttrs),
+    Arr.sort(actualDefinedAttrs)
+  );
+
+  if (!isEqual) {
+    throw new Error('Attribute names were not matching. actual: ' + actualDefinedAttrs.join(', ') + ', expected: ' + allDefinedAttrs.join(', '));
+  }
```

Comment:
If this sorting then asserting is identical functionality for classes, styles, and attributes, perhaps a helper function which does this would be more suitable.
```suggestion
const assertEqualArrays = (expected: T[], actual: T[]): boolean =>  Arr.equal(
  Arr.sort(expected),
  Arr.sort(actual)
);
```